javalib/src/main/scala/java/time/Duration.scala

```diff
@@ -0,0 +1,240 @@
+
+import java.time.temporal._
+
+final class Duration private (seconds: Long, nanos: Int)
+    extends TemporalAmount with Comparable[Duration]
+    with java.io.Serializable {
+
+  import Preconditions.require
+  import Constants._
+  import ChronoUnit._
+
+  require(nanos >= 0 && nanos <= 999999999, "nanos must be >= 0 and <= 999999999")
+
+  private val totalNanos: BigInt = BigInt(seconds) * NANOS_IN_SECOND + nanos
+
+  def get(unit: TemporalUnit): Long = unit match {
+    case SECONDS => seconds
+
+    case NANOS => nanos
+
+    case _ =>
+      throw new UnsupportedTemporalTypeException(s"Unit not supported: $unit")
+  }
+
+  def getUnits(): java.util.List[TemporalUnit] =
+    Seq[TemporalUnit](SECONDS, NANOS).asJava
+
+  def isZero(): Boolean = seconds == 0 && nanos == 0
+
+  def isNegative(): Boolean = seconds < 0
+
+  def getSeconds(): Long = seconds
+
+  def getNano(): Int = nanos
+
+  def withSeconds(seconds: Long): Duration =
+    new Duration(seconds, nanos)
+
+  def withNanos(nanosOfSecond: Int): Duration =
+    new Duration(seconds, nanosOfSecond)
+
+  private def plus(nanos: BigInt): Duration = Duration(totalNanos + nanos)
+
+  def plus(duration: Duration): Duration =
+    plus(BigInt(duration.getSeconds) * NANOS_IN_SECOND + duration.getNano)
+
+  def plus(amount: Long, unit: TemporalUnit): Duration = unit match {
+    case _ if !unit.isDurationEstimated =>
+      plus(unit.getDuration.multipliedBy(amount))
+
+    case DAYS =>
+      plus(unit.getDuration.multipliedBy(amount))
+
+    case _ =>
+      throw new UnsupportedTemporalTypeException(s"Unit not supported: $unit")
+  }
+
+  def plusDays(days: Long): Duration = plus(days, DAYS)
+
+  def plusHours(hours: Long): Duration = plus(hours, HOURS)
+
+  def plusMinutes(minutes: Long): Duration = plus(minutes, MINUTES)
+
+  def plusSeconds(seconds: Long): Duration = plus(seconds, SECONDS)
+
+  def plusMillis(millis: Long): Duration = plus(millis, MILLIS)
+
+  def plusNanos(nanos: Long): Duration = plus(nanos, NANOS)
+
+  def minus(duration: Duration): Duration =
+    plus(BigInt(duration.getSeconds) * -NANOS_IN_SECOND - duration.getNano)
+
+  def minus(amount: Long, unit: TemporalUnit): Duration = unit match {
+    case _ if !unit.isDurationEstimated =>
+      minus(unit.getDuration.multipliedBy(amount))
+
+    case DAYS =>
+      minus(unit.getDuration.multipliedBy(amount))
+
+    case _ =>
+      throw new UnsupportedTemporalTypeException(s"Unit not supported: $unit")
+  }
+
+  def minusDays(days: Long): Duration = minus(days, DAYS)
+
+  def minusHours(hours: Long): Duration = minus(hours, HOURS)
+
+  def minusMinutes(minutes: Long): Duration = minus(minutes, MINUTES)
+
+  def minusSeconds(seconds: Long): Duration = minus(seconds, SECONDS)
+
+  def minusMillis(millis: Long): Duration = minus(millis, MILLIS)
+
+  def minusNanos(nanos: Long): Duration = minus(nanos, NANOS)
+
+  def multipliedBy(multiplicand: Long): Duration =
+    Duration(totalNanos * multiplicand)
```

Comment:
You could do somthing like this to avoid using `BigInt`:
```scala
val mulNanos = Math.multiplyExact(multiplicand, nanos)
val mulSeconds = Math.multiplyExact(multiplicand, seconds)
val newNanos = mulNanos % NANOS_IN_SECOND
val newSeconds = Math.addExact(mulSeconds, mulNanos / NANOS_IN_SCONDS)
new Duration(newSeconds, newNanos)
```
And maybe take care of negative numbers.